libchisel/src/dropsection.rs

```diff
@@ -0,0 +1,172 @@
+
+    #[test]
+    fn keep_intact() {
+        let mut module = builder::module().build();
+        let name = "empty".to_string();
+        let dropper = DropSection::CustomSectionByName(&name);
+        let did_change = dropper.translate_inplace(&mut module).unwrap();
+        assert_eq!(did_change, false);
+    }
+
+    #[test]
+    fn keep_intact_custom_section() {
+        let mut module = builder::module()
+            .with_section(Section::Custom(CustomSection::new(
+                "test".to_string(),
+                vec![],
+            )))
+            .build();
+        let name = "empty".to_string();
+        let dropper = DropSection::CustomSectionByName(&name);
+        let did_change = dropper.translate_inplace(&mut module).unwrap();
+        assert_eq!(did_change, false);
+    }
+
+    #[test]
+    fn remove_custom_section() {
+        let mut module = builder::module()
+            .with_section(Section::Custom(CustomSection::new(
+                "test".to_string(),
+                vec![],
+            )))
+            .build();
+        let name = "test".to_string();
+        let dropper = DropSection::CustomSectionByName(&name);
+        let did_change = dropper.translate_inplace(&mut module).unwrap();
+        assert_eq!(did_change, true);
+    }
+
+    #[test]
+    fn remove_custom_section_by_index() {
+        let mut module = builder::module()
+            .with_section(Section::Custom(CustomSection::new(
+                "test".to_string(),
+                vec![],
+            )))
+            .build();
+        let dropper = DropSection::CustomSectionByIndex(0);
+        let did_change = dropper.translate_inplace(&mut module).unwrap();
+        assert_eq!(did_change, true);
+    }
+
+    #[test]
+    fn remove_oob_custom_section_by_index() {
+        let mut module = builder::module()
+            .with_section(Section::Custom(CustomSection::new(
+                "test".to_string(),
+                vec![],
+            )))
+            .build();
+        let dropper = DropSection::CustomSectionByIndex(1);
+        let did_change = dropper.translate_inplace(&mut module).unwrap();
+        assert_eq!(did_change, false);
+    }
+
+    #[test]
+    fn remove_custom_unknown_by_index() {
+        let mut module = builder::module()
+            .with_section(Section::Custom(CustomSection::new(
+                "test".to_string(),
+                vec![],
+            )))
+            .build();
+        let dropper = DropSection::UnknownSectionByIndex(0);
+        let did_change = dropper.translate_inplace(&mut module).unwrap();
+        assert_eq!(did_change, true);
+    }
+
+    #[test]
+    fn remove_oob_unknown_section_by_index() {
+        let mut module = builder::module()
+            .with_section(Section::Custom(CustomSection::new(
+                "test".to_string(),
+                vec![],
+            )))
+            .build();
+        let dropper = DropSection::UnknownSectionByIndex(1);
```

Comment:
Maybe annotate why this would not mutate.

Reply:
The test name says "oob_unknown_section_index" where oob means out of bounds. Is that enough?

Reply:
It is but at a glance it would be nicer to have some comments or something in the code. It's fine